bin/run-utils.js

```diff
@@ -0,0 +1,192 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+/* eslint-env node */
+
+var Extend = require("lodash").extend;
+var Fs = require("fs-promise");
+var FxProfileFinder = require("firefox-profile/lib/profile_finder");
+var FxRunner = require("fx-runner/lib/run");
+var Path = require("path");
+var When = require("when");
+
+/**
+ * Profiles that do not include "/" are treated as profile names to be used by
+ * the Fx Profile Manager.
+ *
+ * @param {String} profile Name of the profile to test
+ * @return {Boolean}
+ */
+function isProfileName(profile) {
+  if (!profile) {
+    return false;
+  }
+  return !/[\\\/]/.test(profile);
+}
+
+/**
+ * Retrieve the valid path of a profile.
+ *
+ * @param  {String}  profile Name or path of the profile
+ * @return {Promise} Will be resolved with a valid absolute path pointing to the
+ *                   profile or rejected when the path does not exist or when the
+ *                   profile name can not be found.
+ */
+function getProfilePath(profile) {
+  if (isProfileName(profile)) {
+    var finder = new FxProfileFinder();
+    return When.promise(function(resolve, reject) {
+      finder.getPath(profile, function(err, path) {
+        if (err) {
+          reject(err);
+          return;
+        }
+
+        resolve(path);
+      });
+    });
+  }
+
+  return When.promise(function(resolve, reject) {
+    profile = Path.normalize(profile);
+    if (!Path.isAbsolute(profile)) {
+      profile = Path.join(process.cwd(), profile);
+    }
+    Fs.stat(profile).then(function(stat) {
+      if (!stat.isDirectory()) {
+        reject("Profile path '" + profile + "' is not a directory");
+        return;
+      }
+      resolve(profile);
+    }).catch(reject);
+  });
+}
+exports.getProfilePath = getProfilePath;
+
+// Environment variables that make sure that all messages and errors end up in
+// the process stdout or stderr streams.
+var FX_ENV = {
+  "XPCOM_DEBUG_BREAK": "stack",
+  "NS_TRACE_MALLOC_DISABLE_STACKS": "1"
+};
+
+/**
+ * Check a chunk of process output to check if it contains an error message.
+ *
+ * @param  {String} line
+ * @return {Boolean}
+ */
+function isErrorString(line) {
+  return /^\*{25}/.test(line) || /^\s*Message: [\D]*Error/.test(line);
+}
+
+var GARBAGE = [
+  /\[JavaScript Warning: "TypeError: "[\w\d]+" is read-only"\]/,
+  /JavaScript strict warning: /,
+  /\#\#\#\!\!\! \[Child\]\[DispatchAsyncMessage\]/,
+  /JavaScript strict warning: resource:\/\/\/modules\/sessionstore\/SessionStore\.jsm/,
+  /JavaScript strict warning: resource:\/\/gre\/components\/nsSearchService\.js/
```

Comment:
Doesn't the pattern on line 87 cover the ones on 89 and 90, so they're not necessary?